python-checks/src/test/resources/checks/dictionaryDuplicateKey.py

```diff
@@ -0,0 +1,71 @@
+def strings():
+  {"one": 1, "two": 2}
+  {"one": 1, "two": 2, "one": 3}  # Noncompliant {{Change or remove duplicates of this key.}}
+  #^^^^^               ^^^^^<
+  {"one": 1, "two": 2, 'one': 3}  # Noncompliant
+  {"""multi
+  line""": 1, # Noncompliant@-1
+  "two": 2,
+  """multi
+  line""": 3}
+
+  # We only raise if the strings have same values & prefixes, even if they would eventually evaluate to the same value
```

Comment:
I hope it has been discussed elsewhere, because examples like `{"one": 1, "two": 2, u"one": 3}` from the expected-issues currently don't pass.

Reply:
Good point, I don't think the `expected-issues` were updated, but it was discussed indeed and we decided to use the heuristic that the string literals should have the same prefixes and the same literal value when all string elements are considered instead of trying to resolve its actual value. @nicolas-harraudeau-sonarsource  could you please update those when you have time?